frog/imports/client/SingleActivity/index.js

```diff
@@ -263,22 +44,37 @@ class SingleActivity extends React.Component<PropsT, StateT> {
           <ConfigPanel
             activityType={activityType}
             classes={classes}
-            onSubmit={conf =>
-              this.setState({ stage: this.state.stage + 1, activity: conf })
-            }
+            onSubmit={conf => {
```

Comment:
Move Meteor related logic to its own function to separate UI and state concerns.